tools/pythonpkg/src/native/python_objects.cpp

```diff
@@ -434,6 +496,12 @@ py::object PythonObject::FromValue(const Value &val, const LogicalType &type) {
 
 		auto date = val.GetValueUnsafe<date_t>();
 		int32_t year, month, day;
+		if (!duckdb::Date::IsFinite(date)) {
+			if (date == date_t::infinity()) {
+				return py::module::import("datetime").attr("date").attr("max");
```

Comment:
Not caching the min/max values?

Reply:
I don't think our import cache does anything other than remove a few extra string comparisons.
We haven't measured this, but from working with the code for a while I am confident pybind caches the imports itself